static/app/components/assigneeSelectorDropdown.tsx

```diff
@@ -0,0 +1,569 @@
+        </MenuItemFooterWrapper>
+      </InviteMemberLink>
+    );
+  }
+
+  getSuggestedAssignees(): SuggestedAssignee[] {
+    const {suggestedOwners} = this.state;
+    if (!suggestedOwners) {
+      return [];
+    }
+
+    const assignableTeams = this.assignableTeams();
+    const memberList = this.memberList() ?? [];
+    const suggestedAssignees: Array<SuggestedAssignee | null> = suggestedOwners.map(
+      owner => {
+        // converts a backend suggested owner to a suggested assignee
+        const [ownerType, id] = owner.owner.split(':');
+        if (ownerType === 'user') {
+          const member = memberList.find(user => user.id === id);
+          if (member) {
+            return {
+              type: 'user',
+              id,
+              name: member.name,
+              suggestedReason: owner.type,
+              assignee: member,
+            };
+          }
+        } else if (ownerType === 'team') {
+          const matchingTeam = assignableTeams.find(
+            assignableTeam => assignableTeam.id === owner.owner
+          );
+          if (matchingTeam) {
+            return {
+              type: 'team',
+              id,
+              name: matchingTeam.team.name,
+              suggestedReason: owner.type,
+              assignee: matchingTeam,
+            };
+          }
+        }
+
+        return null;
+      }
+    );
+
+    return suggestedAssignees.filter(owner => !!owner) as SuggestedAssignee[];
+  }
+
+  render() {
+    const {disabled, children} = this.props;
+    const {loading, assignedTo} = this.state;
+    const memberList = this.memberList();
+
+    return (
+      <DropdownAutoComplete
+        disabled={disabled}
+        maxHeight={400}
+        onOpen={e => {
+          // This can be called multiple times and does not always have `event`
+          e?.stopPropagation();
+        }}
+        busy={memberList === undefined}
+        items={memberList !== undefined ? this.renderNewDropdownItems() : null}
+        alignMenu="right"
+        onSelect={this.handleAssign}
+        itemSize="small"
+        searchPlaceholder={t('Filter teams and people')}
+        menuFooter={
+          assignedTo ? (
+            <div>
+              <MenuItemFooterWrapper role="button" onClick={this.clearAssignTo} py={0}>
+                <IconContainer>
+                  <IconClose color="purple300" isCircled size="14px" />
+                </IconContainer>
+                <Label>{t('Clear Assignee')}</Label>
+              </MenuItemFooterWrapper>
+              {this.renderInviteMemberLink()}
+            </div>
+          ) : (
+            this.renderInviteMemberLink()
+          )
+        }
+        disableLabelPadding
+        emptyHidesInput
+      >
+        {({getActorProps, isOpen}) =>
+          children({
+            loading,
+            isOpen,
+            getActorProps,
+            assignedTo,
+            suggestedAssignees: this.getSuggestedAssignees(),
```

Comment:
This file is entirely the code from AssigneeSelector with this render prop being the main change. To support the new style of assignee selector